app/controllers/ReportingController.scala

```diff
@@ -110,6 +117,70 @@ trait ReportingController extends BaseController {
     }
   }
 
+  def streamPreviousYearCandidatesDetailsReport: Action[AnyContent] = Action.async { implicit request =>
+    enrichPreviousYearCandidateDetails {
+      (contactDetails, questionnaireDetails, mediaDetails, eventsDetails, siftAnswers) =>
+      {
+        val header = Enumerator(
+          (prevYearCandidatesDetailsRepository.applicationDetailsHeader ::
+            prevYearCandidatesDetailsRepository.contactDetailsHeader ::
+            prevYearCandidatesDetailsRepository.questionnaireDetailsHeader ::
+            prevYearCandidatesDetailsRepository.mediaHeader ::
+            prevYearCandidatesDetailsRepository.eventsDetailsHeader ::
+            prevYearCandidatesDetailsRepository.siftAnswersHeader ::
+            Nil).mkString(",") + "\n"
+        )
+        var counter = 0
+        val candidatesStream = prevYearCandidatesDetailsRepository.applicationDetailsStream().map { app =>
+          val ret = createCandidateInfoBackUpRecord(app, contactDetails, questionnaireDetails, mediaDetails, eventsDetails, siftAnswers) + "\n"
+          // Logger.debug("Output line " + counter + ": " + ret)
+          counter += 1
+          ret
+        }
+        Ok.chunked(Source.fromPublisher(Streams.enumeratorToPublisher(header.andThen(candidatesStream))))
+      }
+    }
+  }
+
+  // scalastyle:off line.size.limit
+  private def enrichPreviousYearCandidateDetails(
+    block: (CsvExtract[String], CsvExtract[String], CsvExtract[String], CsvExtract[String], CsvExtract[String]) => Result
+  ) = {
+    val candidateDetailsFut = prevYearCandidatesDetailsRepository.findContactDetails()
+    val questionnaireDetailsFut = prevYearCandidatesDetailsRepository.findQuestionnaireDetails()
+    val mediaDetailsFut = prevYearCandidatesDetailsRepository.findMediaDetails()
+    val eventsDetailsFut = prevYearCandidatesDetailsRepository.findEventsDetails()
+    val siftAnswersFut = prevYearCandidatesDetailsRepository.findSiftAnswers()
```

Comment:
Any reason we are storing these here? I think they are simple enough to go straight into the for-comprehension

Reply:
👍